// @electron-forge/maker-dmg/dist/MakerDMG.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("source-map-support/register");

var _makerBase = _interopRequireDefault(require("@electron-forge/maker-base"));

var _fsExtra = _interopRequireDefault(require("fs-extra"));

var _path = _interopRequireDefault(require("path"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class MakerDMG extends _makerBase.default {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "name", 'dmg');

    _defineProperty(this, "defaultPlatforms", ['darwin', 'mas']);
  }

  isSupportedOnCurrentPlatform() {
    return process.platform === 'darwin';
  }

  async make({
    dir,
    makeDir,
    appName,
    packageJSON,
    targetArch
  }) {
    // eslint-disable-next-line global-require
    const electronDMG = require('electron-installer-dmg');

    const outPath = _path.default.resolve(makeDir, `${this.config.name || appName}.dmg`);

    const forgeDefaultOutPath = _path.default.resolve(makeDir, `${appName}-${packageJSON.version}-${targetArch}.dmg`);

    await this.ensureFile(outPath);
    const dmgConfig = {
      overwrite: true,
      name: appName,
      ...this.config,
      appPath: _path.default.resolve(dir, `${appName}.app`),
      out: _path.default.dirname(outPath)
    };
    const opts = await electronDMG(dmgConfig);

    if (!this.config.name) {
      await this.ensureFile(forgeDefaultOutPath);
      await _fsExtra.default.rename(outPath, forgeDefaultOutPath);
      return [forgeDefaultOutPath];
    }

    return [opts.dmgPath];
  }

}

exports.default = MakerDMG;
